### PR TITLE
extract determineTransactionType from tx controller

### PR DESCRIPTION
**Pull request by brad-decker**
partially addresses #13422 

- extracts the _determineTransactionType class method from transaction controller. It will additionally be used in the incomingTransactionController too
- no functional changes
- issue states that not putting the method in shared is a requirement, however after careful consideration and confirming that EthQuery, Ethers, etc are already in the frontend bundle this isn't an issue.
- Creates a new string-utils global helper and moves the isEqualCaseInsensitive method to it.

### CODE REVIEW

File: shared/modules/transaction.utils.js [hunk 1]
```diff
@@ -61,3 +81,53 @@ export function txParamsAreDappSuggested(transaction) {
...
+      isContractAddress,
+    } = await readAddressAsContract(query, to);
+
+    contractCode = resultCode;
+    result = isContractAddress
+      ? TRANSACTION_TYPES.CONTRACT_INTERACTION
+      : TRANSACTION_TYPES.SIMPLE_SEND;
+  }
+
+  return { type: result, getCodeResponse: contractCode };
```
Comment (darkwing) [inline shared/modules/transaction.utils.js:132]: I know `getCodeResponse` was used in the previous code but it seems like a weird property name, unless it's a function.

**Review by darkwing — APPROVED** [review]

File: shared/modules/transaction.utils.js [hunk 2]
```diff
@@ -61,3 +81,53 @@ export function txParamsAreDappSuggested(transaction) {
...
+  try {
+    name = data && hstInterface.parseTransaction({ data }).name;
+  } catch (error) {
+    log.debug('Failed to parse transaction data.', error, data);
+  }
+
+  const tokenMethodName = [
+    TRANSACTION_TYPES.TOKEN_METHOD_APPROVE,
+    TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER,
+    TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER_FROM,
```
Comment (adonesky1) [inline shared/modules/transaction.utils.js:108]: also not something we need to do in this PR but I just created this [issue](https://github.com/MetaMask/metamask-extension/issues/13855) and we should consider adding `safeTransferFrom` here soon.

Comment (brad-decker) [reply shared/modules/transaction.utils.js:108]: Yeah, I don't want to make this about anything other than extracting this method. A future PR should be launched to change this behavior.

File: shared/modules/transaction.utils.js [hunk 3]
```diff
@@ -1,4 +1,24 @@
 import { isHexString } from 'ethereumjs-util';
+import { ethers } from 'ethers';
+import abi from 'human-standard-token-abi';
```
[the remark this answers is not in the document]
Comment (brad-decker) [reply shared/modules/transaction.utils.js:3]: In the interest of narrowing scope of this PR I feel we should make that change in a subsequent PR

**Review by DELJUJAZI — APPROVED** [review]

**Review by adonesky1 — APPROVED** [review]
LGTM

**Review by jpuri — APPROVED** [review]
👏   thanks for looking into improving code quality.

**Pull request merged**